File: GfG/overlapping-intervals4919.py
#User function Template for python3
def freturn(ele0):
    return ele0[0]

def overlappedInterval(parr,n):
    parr.sort(key=freturn)
    print("parr:", parr)
    i = 1
    while(i < len(parr)):
        if(parr[i][0] <= parr[i-1][1]):
            parr[i][0] = parr[i-1][0]
            parr[i][1] = max(parr[i-1][1], parr[i][1])
            parr.pop(i-1)
        else:
            i += 1
    return parr


# Driver Code Starts
# Initial Template for Python 3

if __name__=='__main__':
    t= int(input())

    for _ in range(t):
        n=int(input())
        arr=[int(x) for x in input().split()]

        parr=[]

        i=0
        while i<2*n:
            parr.append([arr[i],arr[i+1]])
            i+=2

        ans=overlappedInterval(parr,n)

        for e in ans:
            print(*e,end=' ')
        print()


# Driver Code Ends

File: GfG/overlapping-intervals4919.cpp
// { Driver Code Starts
//Initial function template for C++

#include<bits/stdc++.h>
using namespace std;

vector<pair<int,int>> overlappedInterval(vector<pair<int,int>> , int );

// User code will be pasted here

int main()
{
    
    int t;
    cin>>t;
    while(t--)
    {
        vector<pair<int,int> > arr,res;
        int n,x,y;
        cin>>n;

        for(int i=0;i<n;i++)
        {
            cin>>x>>y;
            arr.push_back(make_pair(x,y));
        }
        
        res = overlappedInterval(arr,n);
        
        
        for(int i=0;i<res.size();i++)
            cout << res[i].first << " " << res[i].second << " " ;
            
        cout<<endl;
    }
}
// } Driver Code Ends


//User function template for C++

void merge(vector<pair<int,int>> vec, int l, int m, int r){
    int i = l, j = m+1, k = 0;
    
}

void mergesort(vector<pair<int,int>> vec, int l, int r){
    if(l < r){
        int m = (l+r)/2;
        mergesort(vec, l, m);
        mergesort(vec, m+1, r);
        merge(vec, l, m, r);
    }
}

vector<pair<int,int>> overlappedInterval(vector<pair<int,int>> vec, int n) {
    
}